include/swift/SIL/SILInstruction.h

```diff
@@ -1222,12 +1222,14 @@ class StringLiteralInst final : public LiteralInst,
 private:
   unsigned Length;
   Encoding TheEncoding;
+  bool NullTerminated;
 
   StringLiteralInst(SILDebugLocation *DebugLoc, StringRef text,
-                    Encoding encoding, SILType ty);
+                    Encoding encoding, bool nullTerminate, SILType ty);
```

Comment:
`SLI->getValue()` should include the null terminator if that's part of the value.